spec/lib/discourse_auth_cookie_spec.rb

```diff
@@ -0,0 +1,246 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+describe DiscourseAuthCookie do
+  fab!(:user) { Fabricate(:user) }
+
+  describe ".parse" do
+    it "does not attempt decryption or validations for v0 cookie" do
+      token = SecureRandom.hex
+      cookie = DiscourseAuthCookie.parse(token)
+      expect(cookie.token).to eq(token)
+    end
+
+    it "decrypts/validates signature for cookies that are < 32 chars" do
+      token = "asdad"
+      expect {
+        DiscourseAuthCookie.parse(token)
+      }.to raise_error(DiscourseAuthCookie::InvalidCookie)
+    end
+
+    it "decrypts/validates signature for v1 cookies" do
+      cookie = DiscourseAuthCookie.new(
+        token: SecureRandom.hex,
+        user_id: user.id,
+        trust_level: user.trust_level,
+        timestamp: 4.hours.ago,
+        valid_for: 100.days
+      ).to_text.dup
+
+      swap1 = 0
+      swap2 = cookie.split("").find_index { |c| c != cookie[swap1] }
+      cookie[swap1], cookie[swap2] = cookie[swap2], cookie[swap1]
```

Comment:
Reading this, it isn't immediately clear what the code is trying to do. Is there a way we can improve the readability of the code here?

Reply:
This finds the first 2 characters in the cookie that are different and swaps them. I could just replace the first and second character, but they happen to be the same in some test run then we end up with the same string and we get a flakey test 😄 If you can think of something simpler I'm open to change it!

Reply:
Maybe just put this in a method and document it with an example?